sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,650 @@
+        """Concatenate and return the predictions of the estimators."""
+        check_is_fitted(self)
+        predictions = [
+            getattr(est, meth)(X)
+            for est, meth in zip(self.estimators_, self.stack_method_)
+            if est != 'drop'
+        ]
+        return self._concatenate_predictions(predictions)
+
+    @if_delegate_has_method(delegate='final_estimator_')
+    def predict(self, X, **predict_params):
+        """Predict target for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        **predict_params : dict of str -> obj
+            Parameters to the `predict` called by the `final_estimator`. Note
+            that this may be used to return uncertainties from some estimators
+            with `return_std` or `return_cov`. Be aware that it will only
+            accounts for uncertainty in the final estimator.
+
+        Returns
+        -------
+        y_pred : ndarray of shape (n_samples,) or (n_samples, n_output)
+            Predicted targets.
+        """
+
+        check_is_fitted(self)
+        return self.final_estimator_.predict(
+            self.transform(X), **predict_params
+        )
+
+
+class StackingClassifier(_BaseStacking, ClassifierMixin):
+    """Stacked of estimators using a final classifier.
+
+    Stacked generalization consists in stacking the output of individual
+    estimator and use a classifier to compute the final prediction. Stacking
+    allows to use the strength of each individual estimator by using their
+    output as input of a final estimator.
+
+    Note that `estimators` are fitted on the full `X` while `final_estimator`
+    is trained using cross-validated predictions of the base estimators using
+    `cross_val_predict`.
+
+    When `predict_proba` is used by each estimator (i.e. most of the time for
+    `stack_method='auto'` or specifically for `stack_method='predict_proba'`)
+    one of the column predicted by each estimator will be dropped since they
+    are collinear.
+
+    .. versionadded:: 0.22
+
+    Read more in the :ref:`User Guide <stacking>`.
+
+    Parameters
+    ----------
+    estimators : list of (str, estimator)
+        Base estimators which will be stacked together. Each element of the
+        list is defined as a tuple of string (i.e. name) and an estimator
+        instance. An estimator can be set to 'drop' using `set_params`.
+
+    final_estimator : estimator, default=None
+        A classifier which will be used to combine the base estimators.
+        The default classifier is a `LogisticRegression`.
+
+    cv : int, cross-validation generator or an iterable, default=None
+        Determines the cross-validation splitting strategy. Possible inputs for
+        cv are:
+
+        * None, to use the default 5-fold cross validation,
```

Comment:
Why don't we just have the default value be 5? I also wonder whether this is one of those cases where a smaller number of splits is good (given sufficient data), because we are not trying to get an estimate of performance, only to get unbiased predictions

Reply:
> Why don't we just have the default value be 5?

We follow the same API than other `EstimatorCV`. However, in all these cases I don't know why `cv=None` is better than `cv=5`. It would be more informative if you don't have the doc.

> I also wonder whether this is one of those cases where a smaller number of splits is good (given sufficient data), because we are not trying to get an estimate of performance, only to get unbiased predictions

It seems reasonable intuitively. We could limit to 3 folds by default.

Reply:
Oh, I see that the CV can change depending of the estimator (e.g. 'gcv' for Ridge). `'auto'` might be better than `None` or even specifying the real default CV. It might due to a reason that I don't know or recall.